build/Extensions/DotNetSettingsExtensions.cs

```diff
@@ -29,13 +28,39 @@ public static DotNetTestSettings SetTargetPlatform(this DotNetTestSettings setti
             : settings.SetProperty("Platform", GetTargetPlatform(platform));
     }
 
+    public static DotNetMSBuildSettings SetTargetPlatform(this DotNetMSBuildSettings settings, MSBuildTargetPlatform platform)
+    {
+        return platform is null
+            ? settings
+            : settings.SetProperty("Platform", GetTargetPlatform(platform));
+    }
+
     public static DotNetTestSettings EnableTrxLogOutput(this DotNetTestSettings settings, string resultsDirectory)
     {
         return settings
             .AddLoggers("trx")
             .SetResultsDirectory(resultsDirectory);
     }
 
+    public static DotNetMSBuildSettings EnableTrxLogOutput(this DotNetMSBuildSettings settings, string resultsDirectory)
```

Comment:
nit: you can consider creating a `*Extensions` class+file per each `Settings` class (or at least reorder the existing extension method definitions so that they are grouped)